packages/@aws-cdk-testing/framework-integ/test/aws-codepipeline-actions/test/cloudformation/integ.stacksets.ts

```diff
@@ -76,7 +76,11 @@ export class StackSetPipelineStack extends Stack {
   }
 }
 
-const app = new App();
+const app = new App({
+  postCliContext: {
+    '@aws-cdk/aws-codepipeline:defaultPipelineTypeToV2': false,
+  },
+});
```

Comment:
If this is not specified, CodeBuild CI will error.

```
@aws-cdk-testing/framework-integ:   CHANGED    pipelines/test/integ.pipeline 1.467s
@aws-cdk-testing/framework-integ:       Resources
@aws-cdk-testing/framework-integ: [~] AWS::CodePipeline::Pipeline Pipeline9850B417 
@aws-cdk-testing/framework-integ:  笏披楳 [+] PipelineType
@aws-cdk-testing/framework-integ:      笏披楳 V2
@aws-cdk-testing/framework-integ: Snapshot Results: 
@aws-cdk-testing/framework-integ: Tests:    41 failed, 801 total
@aws-cdk-testing/framework-integ: Failed: /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/aws-codepipeline-actions/test/cloudformation/integ.stacksets.js
@aws-cdk-testing/framework-integ: Failed: /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/aws-codepipeline-actions/test/integ.cfn-template-from-repo.lit.js
@aws-cdk-testing/framework-integ: Failed: /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/aws-codepipeline-actions/test/integ.lambda-deployed-through-codepipeline.lit.js
@aws-cdk-testing/framework-integ: Failed: /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/aws-codepipeline-actions/test/integ.lambda-pipeline.js
@aws-cdk-testing/framework-integ: Failed: /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/aws-codepipeline-actions/test/integ.pipeline-alexa-deploy.js
@aws-cdk-testing/framework-integ: Failed: /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/aws-codepipeline-actions/test/integ.pipeline-cfn-cross-region.js
@aws-cdk-testing/framework-integ: Failed: /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/aws-codepipeline-actions/test/integ.pipeline-cfn-with-action-role.js
@aws-cdk-testing/framework-integ: Failed: /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/aws-codepipeline-actions/test/integ.pipeline-cfn.js
@aws-cdk-testing/framework-integ: Failed: /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/aws-codepipeline-actions/test/integ.pipeline-code-build-batch.js
@aws-cdk-testing/framework-integ: Failed: /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/aws-codepipeline-actions/test/integ.pipeline-code-build-multiple-inputs-outputs.js
@aws-cdk-testing/framework-integ: Failed: /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/aws-codepipeline-actions/test/integ.pipeline-code-commit-build.js
@aws-cdk-testing/framework-integ: Failed: /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/aws-codepipeline-actions/test/integ.pipeline-code-commit-custom-event.js
@aws-cdk-testing/framework-integ: Failed: /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/aws-codepipeline-actions/test/integ.pipeline-code-commit-main.js
@aws-cdk-testing/framework-integ: Failed: /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/aws-codepipeline-actions/test/integ.pipeline-code-commit.js
@aws-cdk-testing/framework-integ: Failed: /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/aws-codepipeline-actions/test/integ.pipeline-code-deploy-ecs.js
@aws-cdk-testing/framework-integ: Failed: /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/aws-codepipeline-actions/test/integ.pipeline-code-deploy.js
@aws-cdk-testing/framework-integ: Failed: /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/aws-codepipeline-actions/test/integ.pipeline-ecr-source.js
@aws-cdk-testing/framework-integ: Failed: /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/aws-codepipeline-actions/test/integ.pipeline-ecs-deploy.js
@aws-cdk-testing/framework-integ: Failed: /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/aws-codepipeline-actions/test/integ.pipeline-elastic-beanstalk-deploy.js
@aws-cdk-testing/framework-integ: Failed: /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/aws-codepipeline-actions/test/integ.pipeline-events.js
@aws-cdk-testing/framework-integ: Failed: /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/aws-codepipeline-actions/test/integ.pipeline-jenkins.js
@aws-cdk-testing/framework-integ: Failed: /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/aws-codepipeline-actions/test/integ.pipeline-manual-approval.js
@aws-cdk-testing/framework-integ: Failed: /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/aws-codepipeline-actions/test/integ.pipeline-s3-deploy.js
@aws-cdk-testing/framework-integ: Failed: /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/aws-codepipeline-actions/test/integ.pipeline-stepfunctions.js
@aws-cdk-testing/framework-integ: Failed: /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/aws-codepipeline-actions/test/integ.pipeline-with-replication.js
@aws-cdk-testing/framework-integ: Failed: /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/aws-codepipeline-actions/test/s3/integ.source-bucket-events-cross-stack-same-env.js
@aws-cdk-testing/framework-integ: Failed: /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/aws-events-targets/test/codepipeline/integ.pipeline-event-target.js
@aws-cdk-testing/framework-integ: Failed: /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.newpipeline-with-codebuild-logging.js
@aws-cdk-testing/framework-integ: Failed: /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.newpipeline-with-cross-account-keys.js
@aws-cdk-testing/framework-integ: Failed: /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.newpipeline-with-cross-region-replication-buckets.js
@aws-cdk-testing/framework-integ: Failed: /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.newpipeline-with-file-system-locations.js
@aws-cdk-testing/framework-integ: Failed: /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.newpipeline-with-vpc.js
@aws-cdk-testing/framework-integ: Failed: /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.newpipeline.js
@aws-cdk-testing/framework-integ: Failed: /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.pipeline-security.js
@aws-cdk-testing/framework-integ: Failed: /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.pipeline-with-artifact-bucket.js
@aws-cdk-testing/framework-integ: Failed: /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.pipeline-with-assets-single-upload.js
@aws-cdk-testing/framework-integ: Failed: /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.pipeline-with-assets.js
@aws-cdk-testing/framework-integ: Failed: /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.pipeline-with-stack-outputs-in-custom-step.js
@aws-cdk-testing/framework-integ: Failed: /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.pipeline-with-variables.js
@aws-cdk-testing/framework-integ: Failed: /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.pipeline-without-prepare.js
@aws-cdk-testing/framework-integ: Failed: /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.pipeline.js
@aws-cdk-testing/framework-integ: Error: Some tests failed!
@aws-cdk-testing/framework-integ: To re-run failed tests run: integ-runner --update-on-failed
@aws-cdk-testing/framework-integ:     at main (/codebuild/output/src3346000757/src/github.com/aws/aws-cdk/packages/@aws-cdk/integ-runner/lib/cli.js:191:19)
@aws-cdk-testing/framework-integ: Error: integ-runner exited with error code 1
@aws-cdk-testing/framework-integ: Tests failed. Total time (2m5.3s) | integ-runner (1m49.2s) | /codebuild/output/src3346000757/src/github.com/aws/aws-cdk/node_modules/jest/bin/jest.js (14.2s)
@aws-cdk-testing/framework-integ: !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

Reply:
I'm a bit confused why we need to set the feature flag to `false` in context. Wouldn't this check be `false` when no context is provided and doesn't change the integ test behaviour?
```
const isDefaultV2 = FeatureFlags.of(this).isEnabled(cxapi.CODEPIPELINE_DEFAULT_PIPELINE_TYPE_TO_V2);
```

Reply:
I am confused too, I actually got the error in the CodeBuild CI. I thought there was no difference in the local environment...
I'm willing to take time somewhere to investigate.

Reply:
As expected, it appears that if this change is not made, there will be a difference.

https://github.com/aws/aws-cdk/pull/29096/commits/79cd494cbcd1c76874eb504e4c5cfda3f67fb87e

```
@aws-cdk-testing/framework-integ:   CHANGED    aws-codepipeline-actions/test/cloudformation/integ.stacksets 2.077s
@aws-cdk-testing/framework-integ:       Resources
@aws-cdk-testing/framework-integ: [~] AWS::CodePipeline::Pipeline PipelineC660917D 
@aws-cdk-testing/framework-integ:  笏披楳 [+] PipelineType
@aws-cdk-testing/framework-integ:      笏披楳 V2
```

[Build log](https://ops-cdk-autobuild-prs-autobuildv2github-buildlogs-g9jsj1nfmnjp.s3.amazonaws.com/bf7f63f2-d4c5-4d92-a129-3c23e9bb5c8c/build.log?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=ASIAWYORDRBKDGVJQRDI%2F20240312%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240312T151349Z&X-Amz-Expires=600&X-Amz-SignedHeaders=host&X-Amz-Security-Token=IQoJb3JpZ2luX2VjEEAaCXVzLWVhc3QtMSJGMEQCIDjobqLlenwU%2F71O9LMq1L4OzPCAPQTiLp%2Fhy%2F4j02wFAiBJsnUad0qOCM435aTmt1Nyp2gmsqrTenngijU1j8q5lSrTAwhIEAMaDDQ2NDgzMTg3NTE1NiIM%2Bvfr6Js43JSXKbupKrADK4tlo578P528ZBloadTZIH3LACC%2BXey%2FGPDBajyji6MS1qTeIZLKfnSuxxJFmruH5eTFvn7%2FWzH0OJbMgmcYW4SIJJIK1qAUjj5tooZS7xkqokHYwXhKhgoseBlNqGjIJIdYMokPRyFdhp50dYdhxi3GxOtpW8p7WfhcGGf4uZxcUnrS%2BPi3fjuPOGywQ5tg0IY4SgFp8HEViG4Jvh4r9UgRLp2PsrSjHjak5zF9EtI6vlUZGqcM%2B3kxW%2Bwl%2BpbUafrlyMu5rNhfNdy5P2syJCFgt9lMpYy9K8RSIhxV9y8ZC4AN3eFT9J7DyesJ%2FCtemQBhJIYLahCdvjlkXg4wbpZEZm%2BZevYgc82W547K1gTtOJQK1Uc%2FFGLb2M521%2FwJPErlHdRZULjExBOhv5HV7YMo0eU9BlT0%2BZ2Bn2M1R9moJ1tGEY5MZEc7qelac1IRrHpz0roN2RTG1pnjgUZ5SOfu%2BAhpMW7lGBSb6e6TH1IbDSEHIf3bDXWEctpkTW9hyvClBBI%2BbTi9vqSei9rPyKKIysmP8IpuUFQoy%2B7fynLcVcX5AWkmxMVtM%2BKt6ipxMJzhwa8GOp8BOQbRG1sBpNGCBSfDTMzKBSYEHEKJIV3FgF2AY7q4QooTz95lN6DuZ4UykA0M%2B6EBHJAR4Wq0B3omi31tUEXNtRUreBR%2FAqC%2B4XEom2Y0GFcGiCPHLE7jZYtdGiAk3zb8XRR4n8%2Bw%2FcmQmLseaICO6f%2FjllknhBkTqM8HNS6ev9sXOIGzfivXS4akmQKV6Bj0GtEJtaYiFrcmd8XzZG1O&X-Amz-Signature=c2cc3f970c2762ea3870ab6a89e4f531eb815989b673b8f305c3ee159a7f8560)